feature/profile-screen

abdo-essam opened the pull request:
# What does this PR do?
This PR introduces the Account screen, providing users with a dedicated space to manage their profile and application settings.

Key changes include:
- Added `AccountScreen.kt` with its UI layout, including a user profile section and menu items for various account-related actions.
- Implemented `AccountViewModel.kt` to handle the screen's logic and state management using `AccountUiState.kt` and `AccountEffect.kt`.
- Created `AccountContract.kt` to define the interactions between the UI and ViewModel.
- Added new drawable resources for icons used in the Account screen (e.g., `more_vertical.xml`, `user.xml`, `logout.webp`, `time_schedule.webp`).
- Replaced the XML version of `lock_key.xml` with a WebP version and added other WebP icons for design consistency.
- Updated string resources in `values/strings.xml` and `values-ar/strings.xml` to include labels for the new Account screen options.
- Added navigation routes for "Watching History" and "My Rating" in `Screen.kt`.
- Integrated the Account screen into the main app navigation in `NovixApp.kt`.
- Created `UserProfileSection.kt` and `AccountMenuItem.kt` composables for reusable UI components within the Account screen.
- Added new Divider and Dropdown components to the design system.


## Type of Change
- [x] ✨ New feature
- [x] 🎨 UI changes

## Changes Made
- [x] Compose UI

## Screenshots
<img width="761" height="373" alt="Screenshot 2025-08-03 at 2 35 37 PM" src="https://github.com/user-attachments/assets/fd449463-7f49-46a6-ad34-af987a090189" />

<img width="749" height="619" alt="Screenshot 2025-08-03 at 2 37 45 PM" src="https://github.com/user-attachments/assets/e1168c6c-9970-4491-9313-fa31a1b6c6c5" />

abdo-essam committed:
feat(account): Implement account screen UI and basic functionality

Adds account screen with user profile section and menu items.
Includes navigation to watching history and my rating screens.
Adds drawables and strings for account screen elements.
Updates gradle dependencies.

abdo-essam committed:
feat: Implement Account Screen UI

This commit introduces the UI for the account screen, with separate composables for logged-in and not-logged-in states.

- Add `LoggedInContent.kt`:
    - Displays user profile information and a menu with options like Watching History, My Rating, Content Restriction, Change Password, Appearance, and Language.
    - Includes a user menu with a logout option.
    - Uses `UserProfileSection` and `AccountMenuItem` composables.
- Add `NotLoggedInContent.kt`:
    - Shows a message prompting the user to log in.
    - Provides a login button.
    - Displays a placeholder image.

london-bot[bot] committed:
Branch was auto-updated.

abdo-essam committed:
Refactor: Migrate to custom Dropdown and Divider components

- Removed Material 3 dependency from presentation module.
- Replaced Material 3 DropdownMenu and DropdownMenuItem with custom Dropdown and DropdownItem components from the design system.
- Replaced Material 3 HorizontalDivider with custom Divider and SpacerDivider components from the design system.
- Added new Divider and Dropdown components to the design system.

abdo-essam committed:
Merge remote-tracking branch 'origin/feature/profile-screen' into feature/profile-screen

abdo-essam committed:
Refactor: Remove unused Preview code and comments

- Removes unused PreviewParameterProvider and related Composable previews from `AccountScreen.kt`.
- Deletes commented-out lines in `Divider.kt` and `Dropdown.kt`.

abdo-essam committed:
feat: Add user profile image asset

abdo-essam committed:
feat(account): Update AccountScreen UI

AsmaaKhaledGomaa commented on presentation/src/main/java/com/london/presentation/feature/account/components/NotLoggedInContent.kt:
```diff
@@ -0,0 +1,72 @@
+
+@ThemePreviews
+@Composable
+fun NotLoggedInContentPreview() {
```
Please, rename to preview

abdo-essam committed:
feat(account): Make AccountUiState data class properties non-nullable

abdo-essam committed:
feat(account): Set default values for AccountUiState

abdo-essam committed:
refactor(account): Rename preview function for clarity

Abdulrahman-Ragab-01 commented on presentation/src/main/java/com/london/presentation/feature/account/AccountScreen.kt:
```diff
@@ -1,30 +1,100 @@
+    ) {
+        TopBar(
+            title = stringResource(R.string.my_account),
+            modifier = Modifier.padding(vertical = 12.dp),
```
I think the design need more horizontal padding to be like figma design

Abdulrahman-Ragab-01 commented on presentation/src/main/java/com/london/presentation/feature/account/components/UserProfileSection.kt:
```diff
@@ -0,0 +1,146 @@
+        contentAlignment = Alignment.Center
+    ) {
+        Icon(
+            painter = painterResource(dsR.drawable.user),
```
and the user icon as svg then delete the inner alpha to get the original color as in figma

Abdulrahman-Ragab-01 commented on presentation/src/main/java/com/london/presentation/feature/account/components/UserProfileSection.kt:
```diff
@@ -0,0 +1,146 @@
+) {
+    Box {
+        Icon(
+            painter = painterResource(R.drawable.more_vertical),
```
the color of the menu icon in figma is lighter than in application, but its the same color in design system, so I think you can change color from the icon xml itself

Abdulrahman-Ragab-01 commented on presentation/src/main/java/com/london/presentation/feature/account/components/UserProfileSection.kt:
```diff
@@ -0,0 +1,146 @@
+            contentDescription = "Logout",
+            tint = NovixTheme.colors.redAccent,
+            modifier = Modifier
+                .size(20.dp)
```
the icon in figma is larger than in app, I think it's a padding issue

Abdulrahman-Ragab-01 commented on designSystem/src/main/java/com/london/designsystem/component/Dropdown.kt:
```diff
@@ -0,0 +1,61 @@
+}
+
+@Composable
+fun DropdownItem(
```
the drop down item is stuck in the device right frame, but in figma there is a 20 padding in the end

Abdulrahman-Ragab-01 commented on presentation/src/main/java/com/london/presentation/feature/account/components/AccountMenuItem.kt:
```diff
@@ -0,0 +1,94 @@
+        endText?.let {
+            Text(
+                text = it,
+                style = NovixTheme.typography.body.small,
```
style should be label.small

Abdulrahman-Ragab-01 commented on presentation/src/main/java/com/london/presentation/feature/account/components/AccountMenuItem.kt:
```diff
@@ -0,0 +1,94 @@
+            Text(
+                text = it,
+                style = NovixTheme.typography.body.small,
+                color = NovixTheme.colors.body
```
color should be hint

Abdulrahman-Ragab-01 commented on presentation/src/main/java/com/london/presentation/feature/account/components/AccountMenuItem.kt:
```diff
@@ -0,0 +1,94 @@
+            painter = icon,
+            contentDescription = contentDescription,
+            tint = NovixTheme.colors.primary,
+            modifier = Modifier.size(24.dp)
```
size of icon should be 20 enough

abdo-essam replied:
<img width="399" height="161" alt="Screenshot 2025-08-04 at 12 26 24 PM" src="https://github.com/user-attachments/assets/e16607c6-6792-490d-8299-90cc9bba5945" />

in figma it's 24

abdo-essam committed:
style: Adjust dropdown corner radius and logout padding

abdo-essam committed:
refactor: Dropdown component to use Material3 DropdownMenu

london-bot[bot] committed:
Branch was auto-updated.

karimdeabes committed:
Merge remote-tracking branch 'origin/develop' into feature/profile-screen

# Conflicts:
#	presentation/src/main/res/values-ar/strings.xml
#	presentation/src/main/res/values/strings.xml

ahmeddsamii reviewed (commented):
When i continue as guest, this screen appeared? i think this is wrong because i didn't log in

and when i log in my name should appears in this screen or my gmail and this also didn't happened

<img width="403" height="803" alt="image" src="https://github.com/user-attachments/assets/0a3fd496-2672-4e3a-90a8-ed4d2e4523a4" />


logic not implement yet

ahmeddsamii commented on presentation/src/main/java/com/london/presentation/feature/account/AccountViewModel.kt:
```diff
@@ -0,0 +1,78 @@
+
+@HiltViewModel
+class AccountViewModel @Inject constructor(
+    private val context: Application
```
why you need context here? it's not used

and i think it's not allowed to pass context to the viewModel

ahmeddsamii commented on designSystem/src/main/java/com/london/designsystem/component/Divider.kt:
```diff
@@ -0,0 +1,38 @@
+}
+
+@Composable
+fun SpacerDivider(
```
i think we shouldn't use another component to make extra space (16.dp), we should give padding from the component itself

i mean we can use padding of divider itself vertical = 16.dp, it will be better because we gonna save two more component

ahmeddsamii commented on presentation/src/main/java/com/london/presentation/feature/account/AccountEffect.kt:
```diff
@@ -0,0 +1,12 @@
+package com.london.presentation.feature.account
+
+sealed class AccountEffect {
```
this should be sealed interface not class

ahmeddsamii commented on presentation/src/main/java/com/london/presentation/feature/account/AccountScreen.kt:
```diff
@@ -1,30 +1,100 @@
-                style = NovixTheme.typography.headline.medium
+fun AccountScreen(
+    viewModel: AccountViewModel = hiltViewModel(),
+    onNavigateToWatchingHistory: () -> Unit,
```
i think this call back function should be with default value because when any other screen wants to use it and it doesn't want to use these function so it can be, what do you think?

and all other function the same thing

ahmeddsamii commented on presentation/src/main/java/com/london/presentation/feature/account/AccountScreen.kt:
```diff
@@ -1,30 +1,100 @@
+        }
+    }
+
+    AccountScreenContent(
```
why you didn't use BuildScreen Composable? don't you need it?

ahmeddsamii commented on presentation/src/main/java/com/london/presentation/feature/account/AccountScreen.kt:
```diff
@@ -1,30 +1,100 @@
+            .background(NovixTheme.colors.surface)
+            .padding(WindowInsets.statusBars.asPaddingValues())
+    ) {
+        TopBar(
```
this top bar isn't the same horizontal padding with all other screen, please double check this

ahmeddsamii commented on presentation/src/main/java/com/london/presentation/feature/account/components/LoggedInContent.kt:
```diff
@@ -0,0 +1,101 @@
+    uiState: AccountUiState,
+    accountContract: AccountContract
+) {
+    val scrollState = rememberScrollState()
```
you are using this only in one place so you can pass it directly to the verticalScroll()

ahmeddsamii commented on presentation/src/main/java/com/london/presentation/feature/account/components/LoggedInContent.kt:
```diff
@@ -0,0 +1,101 @@
+            showUserMenu = uiState.showUserMenu,
+            onMenuClick = accountContract::onUserMenuClick,
+            onLogoutClick = accountContract::onLogoutClick,
+            modifier = Modifier.padding(top = 12.dp, bottom = 24.dp)
```
this is in figma bottom 40.dp

<img width="703" height="302" alt="image" src="https://github.com/user-attachments/assets/f7404f63-721e-4486-8c82-913ae84c424d" />

abdo-essam replied:
bottom padding for UserProfileSection + vertical padding for AccountMenuItem 
24 + 16 = 40

ahmeddsamii commented on presentation/src/main/java/com/london/presentation/feature/account/components/LoggedInContent.kt:
```diff
@@ -0,0 +1,101 @@
+            onClick = accountContract::onWatchingHistoryClick
+        )
+
+        AccountMenuDivider()
```
you don't need this as i told you before, this is overhead of components, just use the padding of accountMenuItem itself

ahmeddsamii commented on presentation/src/main/java/com/london/presentation/feature/account/components/LoggedInContent.kt:
```diff
@@ -0,0 +1,101 @@
+            onClick = accountContract::onChangePasswordClick
+        )
+
+        AccountMenuDivider()
```
you don't need this as i told you before, this is overhead of components, just use the padding of accountMenuItem itself

ahmeddsamii commented on presentation/src/main/java/com/london/presentation/feature/account/components/LoggedInContent.kt:
```diff
@@ -0,0 +1,101 @@
+}
+
+@Composable
+private fun AccountMenuDivider() {
```
remove this

ahmeddsamii commented on presentation/src/main/java/com/london/presentation/feature/account/components/LoggedInContent.kt:
```diff
@@ -0,0 +1,101 @@
+            onClick = accountContract::onLanguageClick
+        )
+
+        SpacerDivider(thickness = 16.dp)
```
you don't need this as i told you before, this is overhead of components, just use the padding of accountMenuItem itself

abdo-essam replied:
If you add padding inside the AccountMenuItem itself, it will affect all items across the screen that use this component — not just one specific instance

ahmeddsamii commented on presentation/src/main/java/com/london/presentation/feature/account/components/NotLoggedInContent.kt:
```diff
@@ -0,0 +1,72 @@
+            modifier = Modifier.padding(horizontal = 48.dp, vertical = 12.dp)
+        )
+
+        Spacer(modifier = Modifier.height(12.dp))
```
remove this and use padding of components, this is overhead of using extra components

abdo-essam committed:
feat(account): Use Image instead of Icon for profile and menu

abdo-essam committed:
style(account): Update AccountMenuItem text style

abdo-essam committed:
style(account): Change menu item text color to hint

abdo-essam committed:
refactor(account): Remove unused Application context.

abdo-essam committed:
refactor: Remove SpacerDivider composable

Removed SpacerDivider composable, using Spacer instead.

abdo-essam committed:
refactor(account): Use sealed interface for AccountEffect

abdo-essam committed:
feat(account): Add default values to navigation callbacks

abdo-essam committed:
feat(account): Update NotLoggedInContent padding

abdo-essam committed:
refactor(account): Use rememberScrollState directly in Column

london-bot[bot] committed:
Branch was auto-updated.

abdo-essam committed:
feat(account): Integrate BuildScreen and update UI

This commit integrates the BuildScreen for loading and error states and updates the AccountScreen UI.

abdo-essam committed:
Merge remote-tracking branch 'origin/develop' into feature/profile-screen

# Conflicts:
#	presentation/src/main/res/values-ar/strings.xml
#	presentation/src/main/res/values/strings.xml

abdo-essam committed:
feat(account): Replace Image with Icon in UserProfileSection

london-bot[bot] committed:
Branch was auto-updated.

ahmeddsamii reviewed (approved):
Great effort
